### PR TITLE
rg-219: subscription flow. Dependency: rg-240, rg-138

### PR DESCRIPTION
**Pull request by ValentinaTenus**
first page
[
second page
]
3page

### CODE REVIEW

File: backend/src/bundles/payment/payment.service.ts [hunk 1]
```diff
@@ -119,18 +119,66 @@ class PaymentService implements IPaymentService {
...
         }
-
-        const { client_secret } =
+        const { client_secret, payment_method } =
             latest_invoice.payment_intent as Stripe.PaymentIntent;
 
+        const retrievedSubscription = await this.stripe.subscriptions.retrieve(
+            id,
+        );
+        const { current_period_start, current_period_end, items } =
```
Comment (siefimov) [inline backend/src/bundles/payment/payment.service.ts:136]: Names of variables should be written in camelCase notation.
Check it in other places.

File: backend/src/bundles/payment-method/payment-method.model.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    public 'card': string;
+
+    public 'expireDate': Date;
+
+    public static override get tableName(): typeof DatabaseTableName.PAYMENT_METHOD {
+        return DatabaseTableName.PAYMENT_METHOD;
+    }
+
+    public static getRelationMappings(): RelationMappings {
+        return {};
```
Comment (siefimov) [inline backend/src/bundles/payment-method/payment-method.model.ts:22]: Consider adding a relationship for the userId field in the getRelationMappings() method of the PaymentMethodModel class to establish a relationship with the UserModel class.

**Review by siefimov — CHANGES_REQUESTED** [review]

File: backend/src/bundles/payment/helpers/get-card-expire-date.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { type Stripe } from 'stripe';
+
+const getCardExpireDate = (card: Stripe.PaymentMethod.Card): Date => {
+    const { exp_month, exp_year } = card;
+    return new Date(exp_year, exp_month - 1);
```
[the remark this answers is not in the document]
Comment (ValentinaTenus) [reply backend/src/bundles/payment/helpers/get-card-expire-date.ts:5]: from stripe

File: backend/src/bundles/payment/payment.service.ts (hunk 1 above)
Comment (ValentinaTenus) [reply backend/src/bundles/payment/payment.service.ts:136]: from stripe

**Review by siefimov — APPROVED** [review]

**Review by vlad-blip — APPROVED** [review]

**Comment by vlad-blip** [comment]
[Subscription flow (Blocked by #138)](https://trello.com/c/V1g0oF37/219-subscription-flow-blocked-by-138)

**Comment by vlad-blip** [comment]
Conflicts :(

**Pull request closed without merging**